services/streammanager/kafka/client/producer.go

```diff
@@ -140,21 +139,32 @@ func headers(msg Message) (headers []kafka.Header) {
 }
 
 func isErrTemporary(err error) bool {
+	if errors.Is(err, context.DeadlineExceeded) {
+		return true
+	}
```

Comment:
isn't this redundant since we are checking for `Temporary` or `Timeout` errors later on?